src/nvim/getchar.c

```diff
@@ -264,16 +266,16 @@ add_buff (
   buf->bh_index = 0;
 
   ssize_t len;
-  if (buf->bh_space >= (int)slen) {
-    len = STRLEN(buf->bh_curr->b_str);
+  if (buf->bh_space >= slen) {
+    len = (ssize_t)STRLEN(buf->bh_curr->b_str);
     STRLCPY(buf->bh_curr->b_str + len, s, slen + 1);
-    buf->bh_space -= slen;
+    buf->bh_space -= (int)slen;
```

Comment:
Things like this make me uncomfortable. If this cast is ok without checks, why is `slen` not an `int` to begin with?

From the projects perspective, this does not really change anything, because that was a cast `long ->  int` before, which basically has the same problem (maybe not in the platforms nvim runs on, but is that an assumption that should be considered valid forever?).